### PR TITLE
Add Realm.isClose() method.

### LINKED ISSUE #1431: Realm isClosed method
Hi, i noticed there is no method to check my Realm instance is still open/valid. There is no problem when i have RealmObject somewhere because it have "isValid" method so i can call that. But if i have RealmResults or RealmList this validation is not available.
Possible associate with
https://github.com/realm/realm-java/issues/697

### PR DESCRIPTION
**Pull request by zaki50**
This change fixes #1431

### CODE REVIEW

**Comment by zaki50** [comment]
@realm/java review this please

File: realm/src/main/java/io/realm/BaseRealm.java [hunk 1]
```diff
@@ -415,6 +415,19 @@ public void close() {
     }
 
     /**
+     * Check if the {@link io.realm.Realm} instance has already been closed.
```
Comment (beeender) [inline realm/src/main/java/io/realm/BaseRealm.java]: I assume this also works for `DynamicRealm` ? So better to say {@link BaseRealm} here?

Comment (cmelchior) [inline realm/src/main/java/io/realm/BaseRealm.java]: We shouldn't mention BaseRealm it is package protected and only really used as a placeholder for Code. So it is not even part of our JavaDoc

**Comment by cmelchior** [comment]
It needs a unit test for 

```
realm.close()
realm.<randomOperation>
```

But otherwise it looks good.

**Comment by zaki50** [comment]
@cmelchior sure

**Comment by cmelchior** [comment]
Since the unit test is part of #1551.  Dynamic Realm hasn't been merged yet. So the doc update should be part of that PR.

:+1:

**Comment by beeender** [comment]
:+1:

**Pull request merged**